src/cargo/util/toml.rs

```diff
@@ -814,11 +814,14 @@ fn normalize(libs: &[TomlLibTarget],
         for ex in examples.iter() {
             let path = ex.path.clone().unwrap_or_else(|| PathValue::String(default(ex)));
 
-            let profile = Profile::default_test().test(false);
-            let profile = merge(profile, &profiles.test);
+            let profile = merge(Profile::default_example(), &profiles.test);
+            let profile_release = merge(Profile::default_example_release(), &profiles.bench);
```

Comment:
This may want to merge with `profiles.release` instead of `profiles.bench` as you're building it with `--release` instead of `--bench`